src/Host/Client/Impl/Session/RSessionEvaluation.cs

```diff
@@ -27,6 +28,7 @@ internal sealed class RSessionEvaluation : IRSessionEvaluation {
 
         public void Dispose() {
             _hostCancellationRegistration.Dispose();
+            _clientCancellationRegistration.Dispose();
```

Comment:
Since they don't appear to be readonly, is there chance they may be null (i.e. ctor called but code that sets them got never executed)?